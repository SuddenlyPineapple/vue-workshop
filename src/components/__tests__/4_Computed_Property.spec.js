import Vue from "vue";
import Vuetify from "vuetify";
import { mount, createLocalVue } from "@vue/test-utils";
import ComputedProperties from "../4_Computed_Property";

Vue.use(Vuetify);
const localVue = createLocalVue();

describe("4_Computed_Properties", () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const getWrapper = () =>
    mount(ComputedProperties, {
      localVue,
      vuetify
    });

  it("should render computed property component", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should transform user input to uppercase", async () => {
    const wrapper = getWrapper();
    wrapper.find("input").setValue("foo");
    await wrapper.vm.$nextTick()
    expect(wrapper.find(".uppercase").text()).toBe("FOO");
  });
});
