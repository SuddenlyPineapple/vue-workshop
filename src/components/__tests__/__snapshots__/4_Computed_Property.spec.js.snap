// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`4_Computed_Properties should render computed property component 1`] = `
<div><span>Automatically transform text to uppercase:</span>
  <div class="v-input teste theme--light v-text-field">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><input id="input-2" type="text"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
        </div>
      </div>
    </div>
  </div> <span>
    Result:
    <span class="uppercase"></span></span>
</div>
`;
